src/system/RoutesModule/Helper/SanitizeHelper.php

```diff
@@ -13,40 +13,34 @@
 
 namespace Zikula\RoutesModule\Helper;
 
+use function Symfony\Component\String\s;
+
 /**
  * Helper class for sanitizing route properties.
  */
 class SanitizeHelper
 {
+    const SUFFIX_CONTROLLER = 'Controller';
+    const SUFFIX_ACTION = 'Action';
+
     /**
      * Sanitizes the controller / type parameter.
      */
     public function sanitizeController(string $controllerName): array
     {
-        if ('Controller' !== substr($controllerName, -10)) {
-            $type = $controllerName;
-            $controllerName .= 'Controller';
-        } else {
-            $type = substr($controllerName, 0, -10);
-        }
+        $controller = ucfirst(s($controllerName)->ensureEnd(self::SUFFIX_CONTROLLER)->toString());
+        $type = s($controllerName)->trimEnd(self::SUFFIX_CONTROLLER)->lower()->toString();
 
-        $type = strtolower($type);
-        $controllerName = ucfirst($controllerName);
-
-        return [$controllerName, $type];
+        return [$controller, $type];
     }
 
     /**
      * Sanitizes the action / func parameter.
      */
     public function sanitizeAction(string $methodName): array
     {
-        if ('Action' !== substr($methodName, -6)) {
-            $methodName .= 'Action';
-        }
-
-        $methodName = ucfirst($methodName);
-        $func = lcfirst(substr($methodName, 0, -6));
+        $methodName = ucfirst(s($methodName)->ensureEnd(self::SUFFIX_ACTION)->toString());
+        $func = lcfirst(s($methodName)->trimEnd(self::SUFFIX_ACTION)->toString());
```

Comment:
didn't you just remove `Action` from most (all) of our controller methods? should this be changed?

Reply:
I was unsure and decided to review/check this independently as it would include another concern into this PR which is not about changing any behaviour.

Reply:
seems a good approach as long as it is on your radar. Another thought with that is the main controller - selecting controller/action for the main page in general settings-> start page... similar issue in Routes module for creating custom routes. also related is #4376.

Reply:
Collected these things in #4517